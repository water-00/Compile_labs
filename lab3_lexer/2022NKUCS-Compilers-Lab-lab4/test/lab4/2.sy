// i am a function: this function has many syntax errors.

int f() {
	int a;
	a = 0;
	while (a < 10) {
		a = a * 0x11;
		a = a * 011;
		a = a * 11;
	}
	return a;
}

int main() {
	int _testid;
	_testid = 0;
	if(_testid == 0){
		int _testid;
		int _testid2;
		_testid1 = _testid1+1;
		_testid = _testid+1;
	}
	/*
		Comment lines;
	*/
	{
		int a;
	}
	{
		int a;
	}
	int b;
	return 0;
}

/*
	Comment lines.
*/
