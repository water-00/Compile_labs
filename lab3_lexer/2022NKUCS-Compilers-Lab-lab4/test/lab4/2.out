INT	int	3	0	
ID	f	3	4	<0, 1>
LPAREN	(	3	5	
RPAREN	)	3	6	
LBRACE	{	3	8	
INT	int	4	1	
ID	a	4	5	<1, 1>
SEMICOLON	;	4	6	
ID	a	5	1	<1, 1>
ASSIGN	=	5	3	
DECIMAL	0	5	5	
SEMICOLON	;	5	6	
WHILE	while	6	1	
LPAREN	(	6	7	
ID	a	6	8	<1, 1>
LESS	<	6	10	
DECIMAL	10	6	12	
RPAREN	)	6	14	
LBRACE	{	6	16	
ID	a	7	2	<1, 1>
ASSIGN	=	7	4	
ID	a	7	6	<1, 1>
MUL	*	7	8	
HEXADECIMAL	17	7	10	
SEMICOLON	;	7	14	
ID	a	8	2	<1, 1>
ASSIGN	=	8	4	
ID	a	8	6	<1, 1>
MUL	*	8	8	
OCTAL	9	8	10	
SEMICOLON	;	8	13	
ID	a	9	2	<1, 1>
ASSIGN	=	9	4	
ID	a	9	6	<1, 1>
MUL	*	9	8	
DECIMAL	11	9	10	
SEMICOLON	;	9	12	
RBRACE	}	10	1	
RETURN	return	11	1	
ID	a	11	8	<1, 1>
SEMICOLON	;	11	9	
RBRACE	}	12	0	
INT	int	14	0	
ID	main	14	4	<0, 2>
LPAREN	(	14	8	
RPAREN	)	14	9	
LBRACE	{	14	11	
INT	int	15	1	
ID	_testid	15	5	<1, 1>
SEMICOLON	;	15	12	
ID	_testid	16	1	<1, 1>
ASSIGN	=	16	9	
DECIMAL	0	16	11	
SEMICOLON	;	16	12	
IF	if	17	1	
LPAREN	(	17	3	
ID	_testid	17	4	<1, 1>
EQUAL	==	17	12	
DECIMAL	0	17	15	
RPAREN	)	17	16	
LBRACE	{	17	17	
INT	int	18	2	
ID	_testid	18	6	<2, 1>
SEMICOLON	;	18	13	
INT	int	19	2	
ID	_testid2	19	6	<2, 2>
SEMICOLON	;	19	14	
ID	_testid1	20	2	<error>
ASSIGN	=	20	11	
ID	_testid1	20	13	<error>
ADD	+	20	21	
DECIMAL	1	20	22	
SEMICOLON	;	20	23	
ID	_testid	21	2	<2, 1>
ASSIGN	=	21	10	
ID	_testid	21	12	<2, 1>
ADD	+	21	19	
DECIMAL	1	21	20	
SEMICOLON	;	21	21	
RBRACE	}	22	1	
LBRACE	{	26	1	
INT	int	27	2	
ID	a	27	6	<2, 1>
SEMICOLON	;	27	7	
RBRACE	}	28	1	
LBRACE	{	29	1	
INT	int	30	2	
ID	a	30	6	<2, 1>
SEMICOLON	;	30	7	
RBRACE	}	31	1	
INT	int	32	1	
ID	b	32	5	<1, 2>
SEMICOLON	;	32	6	
RETURN	return	33	1	
DECIMAL	0	33	8	
SEMICOLON	;	33	9	
RBRACE	}	34	0	
